scripts/exporter.py

```diff
@@ -85,25 +92,159 @@ def main(self) -> None:
         """Export mesh"""
-        raise NotImplementedError("TSDF mesh reconstruction not implemented yet.")
+
+        if not self.output_dir.exists():
+            self.output_dir.mkdir(parents=True)
+
+        _, pipeline, _ = eval_setup(self.load_config)
+
+        tsdf_utils.export_tsdf_mesh(
+            pipeline,
+            self.output_dir,
+            self.downscale_factor,
+            self.depth_output_name,
+            self.rgb_output_name,
+            self.resolution,
+            self.batch_size,
+            use_bounding_box=self.use_bounding_box,
+            bounding_box_min=self.bounding_box_min,
+            bounding_box_max=self.bounding_box_max,
+        )
+
+        # load the mesh frmo the tsdf export
+        mesh = get_mesh_from_filename(str(self.output_dir / "tsdf_mesh.ply"))
+
+        # possibly
+        # texture the mesh with NeRF and export to a mesh.obj file
+        # and a material and texture file
+        if self.texture_method == "nerf":
+            CONSOLE.print("Texturing mesh with NeRF")
+            texture_utils.export_textured_mesh(mesh, pipeline, self.px_per_uv_triangle, self.output_dir)
 
 
 @dataclass
 class ExportPoissonMesh(Exporter):
     """
-    NOT YET IMPLEMENTED
     Export a mesh using poisson surface reconstruction.
     """
 
+    downscale_factor: int = 2
+    """Downscale factor for the images."""
+    depth_output_name: str = "depth"
+    """Name of the depth output."""
+    rgb_output_name: str = "rgb"
+    """Name of the RGB output."""
+    resolution: Union[int, List[int]] = field(default_factory=lambda: [256, 256, 256])
+    """Resolution of the TSDF volume or [x, y, z] resolutions individually."""
+    batch_size: int = 10
+    """How many depth images to integrate per batch."""
+
     def main(self) -> None:
         """Export mesh"""
-        raise NotImplementedError("Poisson mesh reconstruction not implemented yet.")
+
+        if not self.output_dir.exists():
+            self.output_dir.mkdir(parents=True)
+
+        _, pipeline, _ = eval_setup(self.load_config)
+
+        dataparser_outputs = pipeline.datamanager.train_dataset.dataparser_outputs
+        cameras = dataparser_outputs.cameras
+
+        color_images, depth_images = render_trajectory(
+            pipeline,
+            cameras,
+            rgb_output_name=self.rgb_output_name,
+            depth_output_name=self.depth_output_name,
+            rendered_resolution_scaling_factor=1.0 / self.downscale_factor,
+        )
+
+        # create the point cloud
+        point_cloud = []
+        for i, _ in enumerate(depth_images):
+            ray_bundle = cameras.generate_rays(camera_indices=i)
+            points = ray_bundle.origins + ray_bundle.directions * depth_images[i]
+            point_cloud.append(points)
+        point_cloud = torch.stack(point_cloud)
+        point_cloud = point_cloud.reshape(-1, 3)
+        color_cloud = torch.from_numpy(np.stack(color_images)).reshape(-1, 3)
```

Comment:
This should be replaced with the generate point cloud function. (plus the stuff after this, I couldn't select it all in github)